### PR TITLE
Add validation to pack name is changed

### PR DESCRIPTION
**Pull request by ilaner**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/45255

## Description
Except for special cases, pack name should be unchanged. Added a validation to check if the user changed it, and fail it if so.

## Must have
- [x] Tests
- [x] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/common/errors.py [hunk 1]
```diff
@@ -2104,3 +2105,8 @@ def runas_is_dbotrole():
     @error_code_decorator
     def invalid_yml_file(error):
         return f'There is problem with the yml file. The error: {error}'
+
+    @staticmethod
+    @error_code_decorator
+    def changed_pack_name(original_name, new_name):
+        return f'The pack name cannot be change. change the pack name from {new_name} to {original_name}.'
```
Comment (dorschw) [inline demisto_sdk/commands/common/errors.py:2112]:
```suggestion
        return f'Pack names cannot be changed. Please rename it back to {original_name} under pack_metadata.'
```

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -403,6 +404,21 @@ def run_validation_on_generic_entities(self, dir_path, pack_error_ignore_list):
 
         return all(package_entities_validation_results)
 
+    def is_valid_pack_name(self, file_path, old_file_path):
+        if old_file_path:
+            old_file_path_posix = Path(old_file_path)
+            file_path_posix = Path(file_path)
+            if file_path_posix.parts[0] == old_file_path_posix.parts[0] == 'Packs' and \
```
Comment (dorschw) [inline demisto_sdk/commands/validate/validate_manager.py:411]: do we not chfeck it if not running relatively to `Packs`?

File: demisto_sdk/commands/common/errors.py [hunk 3]
```diff
@@ -2109,4 +2109,4 @@ def invalid_yml_file(error):
     @staticmethod
     @error_code_decorator
     def changed_pack_name(original_name, new_name):
-        return f'The pack name cannot be change. change the pack name from {new_name} to {original_name}.'
+        return f'Pack names cannot be changed. Please rename it back to {original_name} under pack_metadata.'
```
Comment (dorschw) [inline demisto_sdk/commands/common/errors.py:2112]: This string  is incorrect (my bad), please do not merge until we decide on it [here](https://panw-global.slack.com/archives/C02SU1HJZC3/p1641400316000100)

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -2104,3 +2105,8 @@ def runas_is_dbotrole():
     @error_code_decorator
     def invalid_yml_file(error):
         return f'There is problem with the yml file. The error: {error}'
+
+    @staticmethod
+    @error_code_decorator
+    def changed_pack_name(original_name, new_name):
+        return f'Pack names cannot be changed. Please rename it back to {original_name} under pack_metadata.'
```
Comment (dorschw) [inline demisto_sdk/commands/common/errors.py:2112]:
```suggestion
        return f'Pack folder names cannot be changed, please rename it back to {original_name}. If you wish to rename the pack, you can edit the name field in pack_metadata.json, and the pack will be shown in the Marketplace accordingly.'
```

File: demisto_sdk/commands/validate/validate_manager.py (hunk 2 above)
Comment (ilaner) [reply demisto_sdk/commands/validate/validate_manager.py:411]: The path will always be relative to the `git` path.

**Comment by dorschw** [comment]
I'd add a comment next to that line to make it easier on future devs

**Review by DeanArbel — APPROVED** [review]
Nice

**Review by DeanArbel — APPROVED** [review]
Approved again.
Please fix `1 for Unused import`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,7 @@
 * Added support for beta integrations in the **format** command.
 * Fixed an issue where the **postman-codegen** command failed when called with the `--config-out` flag.
 * Removed the integration documentation from the detailed description while performing **split** command to the unified yml file.
+* Add to **validate** a validation for pack name to make sure it is unchanged.
```
Comment (dorschw) [inline CHANGELOG.md:13]:
```suggestion
* Added a test to **validate**, to make sure sure pack paths are not changed.
```

**Pull request merged**